content/when-should-we-automate-tests.md

```diff
@@ -0,0 +1,34 @@
+Title: When should we automate tests
+Date: 2022-01-01
+Category: Posts
+Tags: learning, testing, devops
+Slug: when-should-we-automate-tests
+Author: Aditya Chourasiya
+
+Summary: This article is gist of brainstorming session among QA team(Mohita Kataria, Midhi Naidu and Aditya Chourasiya) at WorkSafeBC
+
+   Automation is advised when tests are **prone to human error** or **complex** or if they need **repeating**, **reporting** , **execution speed** and ofcourse fetch good **return on investment**. 
+   
+   On the conterary, test cases for aesthetics, accessibility, one-time, urgent non-regression(hotfixes), exploratory, with unpredictable results and/or with less return on investment,  should not be automated. 
+   
+<img src="../images/chapline-automation.gif">
+
+# Let's dive in reasons and examples
+
+ These are few reasons to automate (not in order)
+- **Repeating** - Apart from typical regression suite this will include running large number of tests, multi platform, multi browser, performance tests, overnight tests or tests with big set of data.
+- **Critical path** - If the user action sequence / path or application function is critical to your business, it's recommended to automate.
+- **Prone to human error** - large number of data fields, complex technical steps like Database test, Security tests should be automated.
+- **Reporting** - Legal, audit, low confidence or high stake tests should be automated for reporting and logs.
+- **Execution speed** - parallel and automated execution can help run all sort of test much faster, however planning, automation, triage may take more time depending on type of tests.
+- **Return on investment** - While there is no exact number, as a rule of thumb anytime you're saving more money with automation, you should.
+
+ These are few reasons not to automate (not in order)
+- **Aesthetics** - User experience , look and feel, colour and images verification require complex human context and interpretation.
+- **Accessibility** - Similar to aesthetics, accessibility, usablity, language, grammer etc should be verified manually
+- **One-time** - Urgent non-regression hotfix, non-critical, or very small changes can/may be manually verified  
```

Comment:
@adichourasiya Are we sure with this? Would a hotfix not be treated like a feature, in which case we should have automated unit and regression testing?

Reply:
I believe we're saying the same thing, hotfix will have automated unit, integration and UI test part of regression suite, however if these tests are falling outside category of regression, (something quick and dirty, one time verification) , they need not to be automated.